KarhooUISDK/Screens/BookingRequestScreen/FormBookingRequestScreen/CheckoutViews/MoreDetailsView.swift

```diff
@@ -88,36 +89,55 @@ final class MoreDetailsView: UIView {
-            setupCapacityView(forPassenger: true, maxNumber: viewModel.passengerCapacity)
-        }
-        
-        if viewModel.baggageCapacity > 0 {
-            setupCapacityView(forPassenger: false, maxNumber: viewModel.baggageCapacity)
-        }
-        
-        viewModel.fleetCapabilities.forEach { capability in
-            setupCapabilityView(for: capability)
-        }
+        setupVehicleCapacityView(forViewModel: viewModel)
+        setupView(for: viewModel.fleetCapabilities)
         fleetCapabilitiesStackView.addArrangedSubview(detailsLabel)
         
         setupConstraints()
     }
     
-    private func setupCapabilityView(for capability: FleetCapabilities) {
-        let image = capability.image
-        let accessibilityId = KHMoreDetailsViewID.fleetCapabilitiesStackView + "_\(capability.title)"
-        setupView(for: capability.title, with: image, accessibilityId: accessibilityId)
+    private func setupVehicleCapacityView(forViewModel viewModel: QuoteViewModel) {
+        if viewModel.fleetCapabilities.count > 0 {
+            let passengerBaggageStackView = UIStackView()
+            passengerBaggageStackView.accessibilityIdentifier = "container_passenger_baggage_stack"
+            passengerBaggageStackView.translatesAutoresizingMaskIntoConstraints = false
+            passengerBaggageStackView.axis = .horizontal
+            passengerBaggageStackView.distribution = .fillEqually
+            passengerBaggageStackView.alignment = .leading
+            passengerBaggageStackView.spacing = 5
+            
+            if viewModel.passengerCapacity > 0 {
+                let passengerStackView = setupCapacityView(forPassenger: true, maxNumber: viewModel.passengerCapacity)
+                passengerBaggageStackView.addArrangedSubview(passengerStackView)
+            }
+            
+            if viewModel.baggageCapacity > 0 {
+                let baggageStackView = setupCapacityView(forPassenger: false, maxNumber: viewModel.baggageCapacity)
+                passengerBaggageStackView.addArrangedSubview(baggageStackView)
+            }
+            
+            fleetCapabilitiesStackView.addArrangedSubview(passengerBaggageStackView)
+        } else {
+            if viewModel.passengerCapacity > 0 {
+                let passengerStackView = setupCapacityView(forPassenger: true, maxNumber: viewModel.passengerCapacity)
+                fleetCapabilitiesStackView.addArrangedSubview(passengerStackView)
+            }
+            
+            if viewModel.baggageCapacity > 0 {
+                let baggageStackView = setupCapacityView(forPassenger: false, maxNumber: viewModel.baggageCapacity)
+                fleetCapabilitiesStackView.addArrangedSubview(baggageStackView)
+            }
+        }
     }
     
-    private func setupCapacityView(forPassenger passenger: Bool, maxNumber: Int) {
+    private func setupCapacityView(forPassenger passenger: Bool, maxNumber: Int) -> UIStackView {
```

Comment:
I think this would be a good time to tidy up the signature and replace the `forPassenger passenger: Bool` parameter with a CapacityViewType enum